objects/client.c

```diff
@@ -1985,27 +1986,34 @@ client_unmanage(client_t *c, bool window_valid)
  * \param c The client to kill.
  */
 void
-client_kill(client_t *c)
+client_kill(client_t *c, int sig)
 {
-    if(client_hasproto(c, WM_DELETE_WINDOW))
+    if (sig == -1)
     {
-        xcb_client_message_event_t ev;
+        if(client_hasproto(c, WM_DELETE_WINDOW))
+        {
+            xcb_client_message_event_t ev;
 
-        /* Initialize all of event's fields first */
-        p_clear(&ev, 1);
+            /* Initialize all of event's fields first */
+            p_clear(&ev, 1);
 
-        ev.response_type = XCB_CLIENT_MESSAGE;
-        ev.window = c->window;
-        ev.format = 32;
-        ev.data.data32[1] = globalconf.timestamp;
-        ev.type = WM_PROTOCOLS;
-        ev.data.data32[0] = WM_DELETE_WINDOW;
+            ev.response_type = XCB_CLIENT_MESSAGE;
+            ev.window = c->window;
+            ev.format = 32;
+            ev.data.data32[1] = globalconf.timestamp;
+            ev.type = WM_PROTOCOLS;
+            ev.data.data32[0] = WM_DELETE_WINDOW;
 
-        xcb_send_event(globalconf.connection, false, c->window,
-                       XCB_EVENT_MASK_NO_EVENT, (char *) &ev);
+            xcb_send_event(globalconf.connection, false, c->window,
+                        XCB_EVENT_MASK_NO_EVENT, (char *) &ev);
+        }
+        else
+            xcb_kill_client(globalconf.connection, c->window);
     }
     else
-        xcb_kill_client(globalconf.connection, c->window);
+    {
+        kill(c->pid, sig);
```

Comment:
Nope. `c->pid` will be 0 for clients which don't have `_NET_WM_PID` (I just tested with `xcalc`).
`man 2 kill`:
> If pid equals 0, then sig is sent to every process in the process group of the calling process.

Yay, awesome will stop itself if you run xcalc and have the auto-SIGSTOP-"feature" in your config.

Also, just checking `_NET_WM_PID` is not enough since X11 is soooo great at being a remote protocol. Could you please also make this check `WM_CLIENT_MACHINE` and only try to send signals to random processes if the client claims to be coming from the local machine?
https://specifications.freedesktop.org/wm-spec/wm-spec-latest.html#KILLINGWINDOWS